### DISCUSSION TITLE
16 ottobre 2023 - esercizio 1

### DISCUSSION THREAD

**Discussion by IonutCicio** [sapienzastudentsnetwork/basi-di-dati-1#47] [Esami]
![image](https://github.com/sapienzastudentsnetwork/bdi2324/assets/40034429/012159f5-5209-4755-a15c-97236ce8b2f7)

**Comment by Warcophyr** [comment]
![image](https://github.com/sapienzastudentsnetwork/bdi2324/assets/133346488/186983ae-aab2-486a-9c09-011bb9b592d9)

**Comment by niccolo-fato** [comment]
![IMAGE 2024-01-15 08:15:39](https://github.com/sapienzastudentsnetwork/bdi2324/assets/73995334/c1d4c6eb-6d7f-4dff-9768-568f6bc3fd10)

**Reply by Romitoskj** [reply]
Non basterebbe mettere in and le due condizioni (n-pezzi <= 2000 e prezzo_unit <= 500) in un unica selezione per ottenere le tuple che non vogliamo? Alla fine facendo l'intersezione tra due relazioni su cui hai fatto due selezioni è come metterle in and.

**Comment by gioele-m-l** [comment]
1a)
1. $MagMilano = \sigma_{Città="Milano"}(Magazzini)$
2. 
$$Result = \pi_{Prodotti.*, Magazzini.Codice, Magazzini.Indirizzo}(\sigma_{Scorte.N-pezzi>150}(\sigma_{Prezzo_unit>50.00}(Prodotti)\bowtie_{Prodotti.Codice=Scorte.Codice-P}Scorte\bowtie_{Scorte.Codice-M=MagMilano.Codice}MagMilano))$$

1b)
1. $ProdottiLeq500€ = \sigma_{Prezzo\\_unit<=500}(Prodotti)$
2. $ScorteLeq500€ANDLeq2000Pcs = \sigma_{N-Pezzi<=2000}(ProdottiLeq500€\bowtie_{Codice=Codice-P}Scorte)$
3. $MagazziniInvalid = \pi_{Magazzini.*}(Magazzini\bowtie_{Magazzini.Codice=ScorteLeq500€ANDLeq2000Pcs.Codice-M}ScorteLeq500€ANDLeq2000Pcs)$
4. $Result = Magazzini - MagazziniInvalid$
